### ISSUE TITLE
Display Change Log on Update pop up

### ISSUE THREAD

**Issue by royto** [hacs/integration#1248]
## Category

- [ ] Backend/Integration
- [X] Frontend
- [ ] Documentation
- [ ] Bot
- [ ] Other

## Checklist

- [X] This Feature Request only contain 1 request (if you have multiple open multiple feature requests).

## The idea

I am used to look at change log (release notes) before updating an integration, plugin, ...

So i can see if: 
 - there is breaking changes
 - new functionalities i can use
 - fix I'm interested in 

Currently, we have a `change log` button which open the release page (if releases are used).

It could be cool to display change log directly in the pop up so we can see directly changes

## Implementation

![image](https://user-images.githubusercontent.com/6990995/82764411-c6303400-9e0e-11ea-97dd-085b59b9dd17.png)

**Comment by ludeeus** [comment]
There are no common way to show the change log.
That button only works partially, if I where to show it directly, a user would never go to the repo, and information the developer want to show might be lost.
Thanks for the requests, but I will not implement this 👍

**Comment by royto** [comment]
I'm sorry but I don't understand your answer.

In the [github api for latest release](https://developer.github.com/v3/repos/releases/#get-the-latest-release), it contains a `body` property which contains the content of the change log.

Here an example with the latest release of hacs:

![image](https://user-images.githubusercontent.com/6990995/82764732-4fe10100-9e11-11ea-866f-faad7fef098b.png)

The content is in markdown so it is as easy as display readme.md or info.md ...

What do you mean by 

> if I where to show it directly, a user would never go to the repo, and information the developer want to show might be lost.

**Comment by ludeeus** [comment]
Sure, in a perfect world where the users **always** was on the previous version, and **all** repositories use releases that would work.
But sadly that is not the case.

**Comment by royto** [comment]
I agree with your point. 
I don't want to remove the change log button. 

If the repos don't use release, you can display a message indicating `no change log available` for example or even nothing ...

IMHO, display last changes could be a nice addition. If users want to see more change he can still click on the change log button ...

So your remarks make me ask you a question: In that case why do you open the last version of the change log (/releases/1.0.1 for example) and not the release page so the users can see all releases (/releases) he has missed ..

**Comment by ludeeus** [comment]
_maybe_ just _maybe_ it can be done, if the perfect conditions are met.
The button should change.

**Comment by royto** [comment]
Thanks for considering this :thumbsup:

**Comment by vanstinator** [comment]
I had a similar idea. My first thought was also using the release notes. But I agree that that could be fraught with peril given users don't always upgrade reliably and may miss important information.

I'm not familiar with the internals of HACS. But it might be interesting to consider either the `info.md` file or a new file like `releases.md`. You could imagine buckets of release notes that are templated by version so the modal shows the appropriate notes for a user who might be 2 weeks behind on updates.

That's likely more work than just pulling the release notes from GitHub. But it's quite a bit more flexible and powerful. I'd use it on my project.

**Comment by ludeeus** [comment]
That would not be a problem for HACS, but as a maintainer I would never maintain another file for it for my repositories.

**Comment by ludeeus** [comment]
So what I'm thinking is:

if the repository do not use releases, nothing changes.

if the repository uses release, the button does not take you to the latest release, but the release overview.
And releases between the one you have installed, and the latest are displayed like this:
![2020-05-29_17-35-22](https://user-images.githubusercontent.com/15093472/83277932-11619280-a1d3-11ea-85c7-b5e39ee6db71.gif)

**Comment by vanstinator** [comment]
Incredible. I love it.

**Comment by ludeeus** [comment]
Added with https://github.com/hacs/frontend/commit/facca2e4e10daca1a529489e2904bfc2b5148263
